### PR TITLE
Update utils.py to save curl parameter in JSON format.

### PR DESCRIPTION
**Pull request by rick-github**
Dump message in JSON format so that it can be copy&pasted from the logs into the command line for debugging.

### CODE REVIEW

**Review by ishaan-jaff — COMMENTED** [review]
- can you share what the curl command looks like after this change ?

**Comment by rick-github** [comment]
Before:

litellm-1  | POST Request Sent from LiteLLM:
litellm-1  | curl -X POST \
litellm-1  | http://ollama:11434/api/generate \
litellm-1  | -d '{'model': 'mistral', 'prompt': 'why is the sky blue?', 'options': {'temperature': 0.7}, 'stream': False}'

After:

litellm-1  | POST Request Sent from LiteLLM:
litellm-1  | curl -X POST \
litellm-1  | http://ollama:11434/api/generate \
litellm-1  | -d '{"model": "mistral", "prompt": "why is the sky blue?", "options": {"temperature": 0.7}, "stream": false}'

File: litellm/utils.py [hunk 1]
```diff
@@ -1095,7 +1095,7 @@ def pre_call(self, input, api_key, model=None, additional_args={}):
             curl_command += (
                 f"{formatted_headers} \\\n" if formatted_headers.strip() != "" else ""
             )
-            curl_command += f"-d '{str(data)}'\n"
+            curl_command += f"-d '{json.dumps(data)}'\n"
```
Comment (ishaan-jaff) [inline litellm/utils.py]: two requests 
- can we `try/except` the `json.dumps` so it does not block the LLM call 
- can you modify this to not use an `f` string. in python `f` strings are eagerly evaluated and can impact latency

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
left review

File: litellm/utils.py [hunk 2]
```diff
@@ -1090,14 +1090,20 @@ def pre_call(self, input, api_key, model=None, additional_args={}):
...
             verbose_logger.debug(f"PRE-API-CALL ADDITIONAL ARGS: {additional_args}")
-
-            curl_command = "\n\nPOST Request Sent from LiteLLM:\n"
-            curl_command += "curl -X POST \\\n"
-            curl_command += f"{api_base} \\\n"
-            curl_command += (
-                f"{formatted_headers} \\\n" if formatted_headers.strip() != "" else ""
-            )
-            curl_command += f"-d '{str(data)}'\n"
+
```
Comment (ishaan-jaff) [inline litellm/utils.py]: this looks good - but it also means we'll be doing a `json.dumps` per call -> which significantly impacts latency 

can you utilize the value of ` verbose_logger.level` or `litellm.set_verbose` and conditionally run this logic 

```
 formatted_headers_str = "{formatted_headers} \\\n".format(formatted_headers=formatted_headers) if formatted_headers.strip() != "" else ""
            try:
              data_str = json.dumps(data, default=str)
            except:
              data_str = str(data)

            curl_command = (
            "\n\nPOST Request Sent from LiteLLM:\n"
            "curl -X POST \\\n"
            "{api_base} \\\n"
            "{formatted_headers_str}-d '{data_str}'\n"
            ).format(api_base=api_base, formatted_headers_str=formatted_headers_str, data_str=data_str.replace("'", "'\"'\"'"))
```

**Review by ishaan-jaff — CHANGES_REQUESTED** [review]
thanks for the great work so far ! I really want to make sure we don't impact latency for users in production with this change hence added some feedback

File: litellm/utils.py (hunk 2 above)
Comment (rick-github) [reply litellm/utils.py]: Done.

File: litellm/utils.py (hunk 1 above)
Comment (rick-github) [reply litellm/utils.py]: Done.

**Pull request closed without merging**